source/JsMaterialX/test/XmlIo.spec.js

```diff
@@ -1,20 +1,243 @@
-import * as path from 'path';
 import Module from './_build/JsMaterialX.js';
 import { expect } from 'chai';
+import { getMtlxStrings } from './testHelpers';
 
 describe('XmlIo', () => {
-  let mx;
-  before(async () => {
-      mx = await Module();
-  });
-
-  it('should prepend include tag', () => {
-    const doc = mx.createDocument();
-    const includePath = "SomePath";
-    const writeOptions = new mx.XmlWriteOptions();
-    mx.prependXInclude(doc, includePath);
-    const xmlString = mx.writeToXmlString(doc, writeOptions);
-    expect(xmlString).to.include(includePath);
-  });
+    let mx;
```

Comment:
Wow, very comprehensive test suite. I'm wondering if this one is more through than the original cpp and whether it makes sense to put some of these tests there? like the cycle and the deeply nested includes. Could be in another task or PR, just wondering if it makes sense

Reply:
The first three tests are stripped-down versions of a test that is run in the C++ pipeline as well. I haven't checked all of them, but I think there is also one that tests nested includes by now. Some of these JS tests might still be interesting for C++, but it's important to note that we're not testing the C++ implementation here, but the JS one. So even if the tests are duplicated in C++, they should remain here as well.